From 3d1769ffb6abf357f8cf1d6ced3c8bceed0ac198 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@gmail.com>
Date: Thu, 5 May 2022 01:38:53 +0300
Subject: [PATCH 15/98] Don't reset A2DP offload when disabling developer
 settings

* We need to disable A2DP HW offload on FP4 for now, however
  disabling developer settings would enable offload again and
  break SBC and AAC audio. Just remove this section and allow
  this setting to persist after disabling developer settings.

Change-Id: I5b65b3887d68ac684d7ea78de0ee89fb7db9bb0f
---
 ...oothA2dpHwOffloadPreferenceController.java | 19 +++----------------
 1 file changed, 3 insertions(+), 16 deletions(-)

diff --git a/src/com/android/settings/development/BluetoothA2dpHwOffloadPreferenceController.java b/src/com/android/settings/development/BluetoothA2dpHwOffloadPreferenceController.java
index 45d82ad1773..28621bf81cc 100644
--- a/src/com/android/settings/development/BluetoothA2dpHwOffloadPreferenceController.java
+++ b/src/com/android/settings/development/BluetoothA2dpHwOffloadPreferenceController.java
@@ -80,23 +80,10 @@ public class BluetoothA2dpHwOffloadPreferenceController extends DeveloperOptions
         updateState(mPreference);
     }
 
-    @Override
-    protected void onDeveloperOptionsSwitchDisabled() {
-        super.onDeveloperOptionsSwitchDisabled();
-        final boolean offloadSupported =
-                SystemProperties.getBoolean(A2DP_OFFLOAD_SUPPORTED_PROPERTY, false);
-        if (offloadSupported) {
-            ((TwoStatePreference) mPreference).setChecked(false);
-            SystemProperties.set(A2DP_OFFLOAD_DISABLED_PROPERTY, "false");
-        }
-    }
-
     public boolean isDefaultValue() {
-        final boolean offloadSupported =
-                SystemProperties.getBoolean(A2DP_OFFLOAD_SUPPORTED_PROPERTY, false);
-        final boolean offloadDisabled =
-                    SystemProperties.getBoolean(A2DP_OFFLOAD_DISABLED_PROPERTY, false);
-        return offloadSupported ? !offloadDisabled : true;
+        // Always return true here to avoid needing to reboot when disabling
+        // developer options, since we aren't turning this off when doing so anymore.
+        return true;
     }
 
     /**
